sound/soc/sof/ipc4-topology.c

```diff
@@ -853,10 +912,15 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 		rate = fmt->sampling_frequency;
 		channels = SOF_IPC4_AUDIO_FORMAT_CFG_CHANNELS_COUNT(fmt->fmt_cfg);
 		valid_bits = SOF_IPC4_AUDIO_FORMAT_CFG_V_BIT_DEPTH(fmt->fmt_cfg);
-		if (params_rate(params) == rate && params_channels(params) == channels &&
-		    sample_valid_bits == valid_bits) {
-			dev_dbg(sdev->dev, "matching audio format index for %uHz, %ubit, %u channels: %d\n",
-				rate, valid_bits, channels, i);
+
+		/*
+		 * use SNDRV_PCM_RATE_8000_192000 to avoid
+		 * dozens of format setting in topology for SRC case
+		 */
+		if ((params_rate(params) == rate || rate == SOF_IPC4_SUPPORTED_PCM_RATE) &&
```

Comment:
Better @RanderWang but you missed part of the feedback: what tells you that the SRC supports all conversions from any source rate to any destination rate? There's got to be something, possibly in the topology or module description, that tells us what we can do with the mixer.

In addition, some modules have very strict requirements, you cannot e.g. connect an input to a mixer if its rate is different. I worry that we are giving a blanket permission to everyone only to deal with the specific SRC case?

Reply:
In my mind everything should be defined by topology.  there are two cases I can imagine: (1）playback with different rates of pcm. (2) use SRC to convert rate in pipeline.  For (1) we do SRC after host copier and convert rate to some value, then we don't need other modules to deal with different rates. For case (2) the source and sink rate should be defined to a fixed value, so we don't need to use SOF_IPC4_SUPPORTED_PCM_RATE in topology, just use designed rate

Reply:
I can remove SOF_IPC4_SUPPORTED_PCM_RATE in driver and list every supported rates in topology for SRC and host Copier,  but there are about 30+ format entries for each module, almost a few screen pages.

Reply:
It is better to make driver simple. Now update topology which take care of SRC support

Reply:
Can you point us to the topology update? Thanks!

Reply:
sure, https://github.com/thesofproject/sof/pull/5913/commits/2af0d94b6a5db0c4a373ce8f0e8315cddf412543